plugin/src/main/java/org/opensearch/ml/action/memorycontainer/memory/TransportDeleteEventAction.java

```diff
@@ -0,0 +1,113 @@
+import lombok.extern.log4j.Log4j2;
+
+/**
+ * Transport action for deleting an event from a memory container
+ */
+@Log4j2
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class TransportDeleteEventAction extends HandledTransportAction<MLDeleteEventRequest, DeleteResponse> {
+
+    final Client client;
+    final MLFeatureEnabledSetting mlFeatureEnabledSetting;
+    final MemoryContainerHelper memoryContainerHelper;
+
+    @Inject
+    public TransportDeleteEventAction(
+        TransportService transportService,
+        ActionFilters actionFilters,
+        Client client,
+        MLFeatureEnabledSetting mlFeatureEnabledSetting,
+        MemoryContainerHelper memoryContainerHelper
+    ) {
+        super(MLDeleteEventAction.NAME, transportService, actionFilters, MLDeleteEventRequest::new);
+        this.client = client;
+        this.mlFeatureEnabledSetting = mlFeatureEnabledSetting;
+        this.memoryContainerHelper = memoryContainerHelper;
+    }
+
+    @Override
+    protected void doExecute(Task task, MLDeleteEventRequest request, ActionListener<DeleteResponse> actionListener) {
+        if (!mlFeatureEnabledSetting.isAgenticMemoryEnabled()) {
+            actionListener.onFailure(new OpenSearchStatusException(ML_COMMONS_AGENTIC_MEMORY_DISABLED_MESSAGE, RestStatus.FORBIDDEN));
+            return;
+        }
+
+        User user = RestActionUtils.getUserContext(client);
+        String memoryContainerId = request.getMemoryContainerId();
+        String eventId = request.getEventId();
+
+        if (StringUtils.isBlank(memoryContainerId)) {
+            actionListener.onFailure(new IllegalArgumentException("Memory container ID is required"));
+            return;
+        }
+
+        if (StringUtils.isBlank(eventId)) {
+            actionListener.onFailure(new IllegalArgumentException("Event ID is required"));
+            return;
+        }
+
+        memoryContainerHelper.getMemoryContainer(memoryContainerId, ActionListener.wrap(container -> {
+            if (!memoryContainerHelper.checkMemoryContainerAccess(user, container)) {
+                actionListener
+                    .onFailure(
+                        new OpenSearchStatusException(
+                            "User doesn't have permissions to delete from this memory container",
+                            RestStatus.FORBIDDEN
+                        )
+                    );
+                return;
+            }
+
+            deleteEventFromIndex(eventId, container.getConfiguration(), actionListener);
+        }, actionListener::onFailure));
+    }
+
+    private void deleteEventFromIndex(String eventId, MemoryConfiguration configuration, ActionListener<DeleteResponse> actionListener) {
+        try {
+            String eventIndexName = configuration.getShortTermMemoryIndexName();
+            DeleteRequest deleteRequest = new DeleteRequest(eventIndexName, eventId);
+
+            client.delete(deleteRequest, ActionListener.wrap(deleteResponse -> {
```

Comment:
Since we store memory to system index by default if user not provide index prefix. We should check if the index is system index or not. If it's system index, we should check user's permission and stash context.

The memory should be user level. Memory container should support multiple users' memories. So only checking if user have memory container access is not enough. We should also record user id in memory index and add filter when search memory.